include/boost/geometry/algorithms/detail/distance/geometry_to_segment_or_box.hpp

```diff
@@ -220,7 +220,8 @@ class geometry_to_segment_or_box
         comparable_return_type cd_min1(0);
         point_iterator_type pit_min;
         seg_or_box_iterator_type it_min1 = seg_or_box_points.begin();
-        seg_or_box_iterator_type it_min2 = ++seg_or_box_points.begin();
+        seg_or_box_iterator_type it_min2 = it_min1;
+        ++it_min2;
```

Comment:
Ok, I'll take a look at it later. Using const iterators explicitly shoudl of course be safer, though this is a different change. Have in mind that the conversion from mutable to const iterator is not a problem here and below. This is a different issue than the one I explained.

Here the compiler complains about calling pre-increment or post-increment operator for temporary objects. In general this works when an operator is a member function of a class. But wouldn't work e.g. if the iterator was a pointer, or if the user-defined operator was a free function because rvalue couldn't be converted to non-const reference. And AFAIU on Solaris Compiler it doesn't work in any case.